### PR TITLE
Feat: add support for update credentials

### LINKED ISSUE #678: `env0_azure_cost_credentials` Update Credentials Instead of Replace
Using provider `1.4.9`.

It'd be good to update credentials with a new secret instead of replace, example:

```
# module.env0-credentials.env0_azure_cost_credentials.default["###"] must be replaced
-/+ resource "env0_azure_cost_credentials" "default" {
      ~ client_id       = (sensitive value) # forces replacement
      ~ client_secret   = (sensitive value) # forces replacement
      ~ id              = "###" -> (known after apply)
        name            = "###"
        # (2 unchanged attributes hidden)
    }
```

I have a separate terraform state to manage projects and their cost credential associations, etc.., so there's a cross run dependency now on this when the credential is updated. I can accomodate this with the env0 workflow trigger as a work around.

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
Resolves #678 

### Solution

Part of this PR was upgrading to go 1.20 and replace gomock location (original project has been deprecated).
I have also updated github action versions. Therefore, "87" files are changed.

Some guidelines for reviewing this PR:
1. client/cloud_credentials.go - added the new API for updating credentials.
2. env0/resource_cost_credentials.go - was refactored, and the update operation was added.
3. env0/resource_cost_credentials_test.go - acceptance tests were updated.
4. The rest of the files are minor changes.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ env:
   ENV0_API_ENDPOINT: ${{ secrets.ENV0_API_ENDPOINT }}
   ENV0_API_KEY: ${{ secrets.TF_PROVIDER_INTEGRATION_TEST_API_KEY }} # API Key for organization 'TF-provider-integration-tests' @ dev
   ENV0_API_SECRET: ${{ secrets.TF_PROVIDER_INTEGRATION_TEST_API_SECRET }}
-  GO_VERSION: 1.19
+  GO_VERSION: "1.20"
```
Comment (TomerHeber) [inline .github/workflows/ci.yml:11]: had to add quotes. There's a bug/issue if this isn't done. (Will use go 1.2).

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -18,14 +18,14 @@ jobs:
...
-        uses: actions/setup-go@v3
+        uses: actions/setup-go@v4
         with:
           go-version: ${{ env.GO_VERSION }}
       - name: Checkout code
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
       - name: Generate mocks
         run: |
-          go install github.com/golang/mock/mockgen@v1.6.0
```
Comment (TomerHeber) [inline .github/workflows/ci.yml:28]: This has been deprecated. Ownership moved to Uber.

File: client/cloud_credentials.go [hunk 3]
```diff
@@ -143,10 +143,27 @@ func (client *ApiClient) CredentialsCreate(request CredentialCreatePayload) (Cre
 	request.SetOrganizationId(organizationId)
 
 	var result Credentials
-	err = client.http.Post("/credentials", request, &result)
+	if err := client.http.Post("/credentials", request, &result); err != nil {
```
Comment (TomerHeber) [inline client/cloud_credentials.go:146]: github diff mixup... this did not change other than this line.

File: client/cloud_credentials.go [hunk 4]
```diff
@@ -143,10 +143,27 @@ func (client *ApiClient) CredentialsCreate(request CredentialCreatePayload) (Cre
...
+		return Credentials{}, err
+	}
+
+	return result, nil
+}
+
+func (client *ApiClient) CredentialsUpdate(id string, request CredentialCreatePayload) (Credentials, error) {
```
Comment (TomerHeber) [inline client/cloud_credentials.go:153]: this is all new

File: client/cloud_credentials_test.go [hunk 5]
```diff
@@ -115,6 +115,46 @@ var _ = Describe("CloudCredentials", func() {
 		})
 	})
 
+	Describe("AwsCredentialsUpdate", func() {
```
Comment (TomerHeber) [inline client/cloud_credentials_test.go:118]: unit test for the new update API.

File: env0/resource_cost_credentials.go [hunk 6]
```diff
@@ -2,7 +2,6 @@ package env0
```
Comment (TomerHeber) [inline env0/resource_cost_credentials.go:2]: This code has been refactored. Easier to support the Update use-case without duplicating code.

Comment (TomerHeber) [reply env0/resource_cost_credentials.go:2]: might be easier to just review the whole file.

File: go.mod [hunk 7]
```diff
@@ -1,12 +1,11 @@
 module github.com/env0/terraform-provider-env0
 
-go 1.19
+go 1.20
 
 require (
 	github.com/Masterminds/semver/v3 v3.2.1
 	github.com/adhocore/gronx v1.6.5
 	github.com/go-resty/resty/v2 v2.7.0
-	github.com/golang/mock v1.6.0
```
Comment (Wassap124) [inline go.mod:9]: why did you swap `github.com/golang/mock/gomock` for `go.uber.org/mock`

Comment (TomerHeber) [reply go.mod:9]: github.com/golang/mock/gomock has been deprecated.
go.uber.org/mock replaces it.

See: https://github.com/golang/mock

**Review by Wassap124 — APPROVED** [review]

**Pull request merged**